source/FreeRTOS_IP.c

```diff
@@ -387,6 +387,7 @@ static void prvProcessIPEventsAndTimers( void )
 
                     /* Cast in two steps to please MISRA. */
                     uxState = ( uintptr_t ) xReceivedEvent.pvData;
+                    /* coverity[misra_c_2012_rule_10_5_violation] */
```

Comment:
If we are suppressing this rule anyways, then why not remove line 389? It seems that the only purpose of that line was to "please MISRA"

Reply:
this violates 2 misra rules
the first line removes one, and the comment another one

Reply:
@alfred2g wrote:

> the first line removes one, and the comment another one

Exactly, it was indeed a way to please MISRA:
~~~c
    uintptr_t uxState = ( uintptr_t ) xReceivedEvent.pvData;
    eState = ( eDHCPState_t ) uxState;
~~~
The `pvData` field already exists for many years, it was there in the +UDP library version 1.0 ten years ago. It is a void pointer used for anything.